test/company_schema.js

```diff
@@ -16,4 +18,41 @@ describe("# Company Schema tests", () => {
             CompanySchemaTester.maxLength("name", CompanyConstants.companyName.max_length);
         });
     });
+
+    describe("Hook tests", () => {
+        let company;
+
+        beforeAll(async () => {
+            company = await Company.create({
+                name: "first name"
+            });
+            const offer = {
+                title: "Test Offer",
+                publishDate: new Date(Date.now() - (DAY_TO_MS)),
+                publishEndDate: new Date(Date.now() + (DAY_TO_MS)),
+                description: "For Testing Purposes",
+                contacts: ["geral@niaefeup.pt", "229417766"],
+                jobType: "SUMMER INTERNSHIP",
+                fields: ["DEVOPS", "MACHINE LEARNING", "OTHER"],
+                technologies: ["React", "CSS"],
+                location: "Testing Street, Test City, 123",
+                owner: company._id,
+                ownerName: company.name
+            };
+
+            await Offer.create([offer, offer]);
+        });
+
+        afterAll(async () => {
+            await Offer.deleteMany({});
+            await Company.deleteMany({});
+        });
```

Comment:
This should probably be inside another describe since it will not be a shared state for all "Hook tests".